src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1645,6 +1645,34 @@ public bool Equals(StringBuilder sb)
             }
         }
 
+        /// <summary>
+        /// Determines if the contents of this builder are equal to the contents of ReadOnlySpan<char>.
+        /// </summary>
+        /// <param name="value">The ReadOnlySpan<char>.</param>
+        public bool Equals(ReadOnlySpan<char> value)
+        {
+            if (value.Length != Length)
+                return false;
+
+            StringBuilder thisChunk = this;
+            int offset = 0;
+            int chunk_length = -1;
+            
+            
+            while (thisChunk != null)
+            {
+                chunk_length = thisChunk.m_ChunkLength;                 
+                offset += chunk_length;
+
+                if(!StringSpanHelpers.Equals(thisChunk.m_ChunkChars, value.Slice(value.Length - offset, chunk_length)))
```

Comment:
What if the whole of m_ChunkChars isn't used?

Reply:
```
 ReadOnlySpan<char> chunk = new ReadOnlySpan<char>(sbChunk.m_ChunkChars, 0, chunk_length);
if (!StringSpanHelpers.Equals(chunk, value.Slice(value.Length - offset, chunk_length)))
                    return false;
```
how about this ?

Reply:
Yup

Reply:
We can even do this once outside the loop if we dont want to allocate memory in each loop.(whole of m_chars wont be used in first case)

Reply:
Do what outside the loop? Each iteration through the loop is for a different chunk.

Reply:
(and creating a span doesn't allocate)

Reply:
```
            int offset = 0;
            int chunk_length = sbChunk.m_ChunkLength;
            offset += chunk_length;

            ReadOnlySpan<char> chunk = new ReadOnlySpan<char>(sbChunk.m_ChunkChars, 0, chunk_length);

            if (!StringSpanHelpers.Equals(chunk, value.Slice(value.Length - offset, chunk_length)))
                return false;

            sbChunk = sbChunk.m_ChunkPrevious;

            while (sbChunk != null)
            {
                chunk_length = sbChunk.m_ChunkLength;
                offset += chunk_length;
    
                if (!StringSpanHelpers.Equals(sbChunk.m_ChunkChars, value.Slice(value.Length - offset, chunk_length)))
                    return false;              
            } 

            Debug.Assert(offset == Length);

            return true;
```
This way we wont have to allocate the memory for readonlyspan each time in loop.